pkg/upgrademonitor/upgrade_monitor.go

```diff
@@ -180,19 +185,16 @@ func generateAndApplyMachineConfigNodes(
 				}
 				newC.DeepCopyInto(&newMCNode.Status.Conditions[i])
 			} else if newChildCondition != nil && condition.Type == newChildCondition.Type {
-				childDNEOrIsTheSame = false
```

Comment:
the value of  childDNEOrIsTheSame could only ever be changed here but then could never be consumed.